can/interfaces/vector/canlib.py

```diff
@@ -295,11 +300,22 @@ def __init__(
         # Calculate time offset for absolute timestamps
         offset = xlclass.XLuint64()
         try:
-            try:
-                xldriver.xlGetSyncTime(self.port_handle, offset)
-            except VectorError:
-                xldriver.xlGetChannelTime(self.port_handle, self.mask, offset)
-            self._time_offset = time.time() - offset.value * 1e-9
+            if time.get_clock_info("time").resolution > 1e-5:
```

Comment:
There is a lot of code duplication. Why not wrapping it in an object like this in can_util:
```
class PreciseTime:
    def __init__(self):
            # use this if the resolution is higher than 10us
            if get_clock_info("time").resolution > 1e-5:
                t0 = time()
                while True:
                    t1, performance_counter = time(), perf_counter()
                    if t1 != t0:
                        break
                self.time = t1
                self.perfcounter = performance_counter
            else:
                self.time = time()
                self.perfcounter = None # not needed

    def now(self):
        if self.perf_counter is None:
            return time()
        return self.time + (perf_counter() - self.perfcounter)
```
Then you need no extra if to you use it, you could just write:
```
            precisetime = PreciseTime()
            try:
                xldriver.xlGetSyncTime(self.port_handle, offset)
            except VectorError:
                xldriver.xlGetChannelTime(self.port_handle, self.mask, offset)
            self._time_offset = precisetime.now() - offset.value * 1e-9
```

Reply:
BTW: the constructor could have also a param required_resolution instead of hard coding 1e-5